src/modules/cmd_duration.rs

```diff
@@ -64,12 +65,24 @@ fn undistract_me<'a, 'b>(
 fn undistract_me<'a, 'b>(
     module: Module<'a>,
     config: &'b CmdDurationConfig,
+    context: &'a Context,
     elapsed: u128,
 ) -> Module<'a> {
     use notify_rust::{Notification, Timeout};
     use nu_ansi_term::{unstyle, AnsiStrings};
 
     if config.show_notifications && config.min_time_to_notify as u128 <= elapsed {
+        if cfg!(target_os = "linux") {
+            let in_graphical_session = ["DISPLAY", "WAYLAND_DISPLAY", "MIR_SOCKET"]
+                .iter()
+                .find_map(|&var| context.get_env(var).filter(|val| !val.is_empty()))
```

Comment:
I think this can just be
```suggestion
                .find_map(|&var| context.get_env(var).is_some())
```

Reply:
Can't we get a false positive for something like `DISPLAY=` , where the variable is set but the value is currently empty? Does `context.get_env` already check this? Might be a silly edge case but i just want to signal that the length check is intentional.

Reply:
IIRC on Windows it isn't possible to differentiate between empty and missing environment variables, at least in some cases.

Reply:
In our case, will that present as an empty variable or a `None` variable?

At any rate, that probably won't affect this PR directly because this change is `cfg!`-ed to Linux-only.

Reply:
I can find at least a few cases where `DISPLAY` is empty but set on the Arch forums. I'm not certain that these cases would impact us (as it appears that it's impossible to launch graphical programs like terminal emulators when this happens), but I think I'm okay with keeping the current code just in case. @andytom Would you object to merging in the current state?

Reply:
Yep no problem from me we can always refactor later if we want.